src/BootApp.ts

```diff
@@ -8,6 +8,16 @@ import { ClassPathData, MainClassData } from "./types/jdtls";
 import { isActuatorJarFile, isAlive } from "./utils";
 import { BootAppItem } from "./views/items/BootAppItem";
 import * as lsp from "vscode-languageclient";
+import * as async from "async";
+import { cpus } from "os";
+
+const searchQueue = async.queue(async (path: string, callback: async.AsyncResultCallback<string[], Error>) => {
+    try {
+        callback(undefined, await vscode.commands.executeCommand('java.execute.workspaceCommand', 'vscode.java.resolveMainClass', path) as string[]);
+    } catch (error) {
+        callback(error);
+    }
+}, cpus().length / 2);
```

Comment:
Use `Math.max(1, cpus().length / 2)` to ensure at least 1 queue for run.

Reply:
Agreed :-)